src/utils/static-eval.js

```diff
@@ -69,7 +69,7 @@ const visitors = {
       }
       // A || UNKNOWN -> A if A is truthy
       if (!('test' in l) && op === '||' && l.value)
-        return node.right;
+        return l;
```

Comment:
This fix is correct, but I'm not sure I follow how it makes the test case work since `({}[chunkId] || chunkId)` should effectively be a case of `undefined || UNKNOWN` (or possibly `UNKNOWN || UNKNOWN` if we want to support prototype properties).

The `undefined || UNKNOWN` case we can handle, the `UNKNOWN || UNKNOWN` case would likely require us to distinguish between a `UNKNOWN_WITH_SIDE_EFFECT` and `UNKNOWN_PURE`. That seems the ideal analysis fix, but I would suggest perhaps just going with `undefined || UNKNOWN` for now. This case should fall out of the standard logical analysis fine though I believe (as opposed to this path which is somewhat of a shortcut inlining the fact that the `UNKNOWN` statement in `A || UNKNOWN` is side-effect free by definition).

So that means that we need to ensure the analysis is returning `undefined` for `{}[chunkId]` (and not `UNKNOWN` or `WILDCARD` as it currently does in this PR).

Reply:
In this case it was making it work by returning the `WILDCARD` for the visited member expression on the left hand side (`{}[unknownComputedProp]`), but as you said, we don't want to use `WILDCARD`!

Either way, this fix is still correct! 😄